### PR TITLE
Gevariah/ota e2e tests fixes release 1.5

### PR DESCRIPTION
**Pull request by HarnishaA**
Updating the bootloader files only for the boards needed. Fixed the test results in case of failure.

Description
-----------
Updating the bootloader files only for the boards needed. Fixed the test results in case of failure it was not getting marked as FAIL in the xml in all the scenarios.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py [hunk 1]
```diff
@@ -154,19 +155,21 @@ def buildProject(self):
...
+                os.path.join(self._projectRootDir, OtaAfrProject.OTA_BOOTLOADER_CONFIG_PATH)
+            )
 
     def copyCodesignerCertificateToBootloader(self, certificate):
         """Copies the input certificate to a file named: aws_ota_codesigner_certificate.pem under
         demos/ota/bootloader/utility/codesigner_cert_utility.
         """
-        with open(os.path.join(self._projectRootDir, OtaAfrProject.OTA_BOOTLOADER_CERTIFICATE_PATH), 'w') as f:
-            f.write(certificate)
+        if OtaAfrProject.OTA_BOOTLOADER_CERTIFICATE_PATH is not None:
```
Comment (azlwwan) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py]: "is not None" is not required.

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py [hunk 2]
```diff
@@ -154,19 +155,21 @@ def buildProject(self):
 
     def __incrementBootloaderSequenceNumber(self):
         self._bootloaderSequenceNumber += 1
-        self.__setIdentifierInFile(
-            {
-                'SEQUENCE_NUMBER': '= ' + str(self._bootloaderSequenceNumber)
-            },
-            os.path.join(self._projectRootDir, OtaAfrProject.OTA_BOOTLOADER_CONFIG_PATH)
-        )
+        if OtaAfrProject.OTA_BOOTLOADER_CONFIG_PATH is not None:
```
Comment (azlwwan) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py]: "is not None" is not required.

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py [hunk 3]
```diff
@@ -100,7 +101,7 @@ def initializeOtaProject(self):
     def generateFactoryImage(self):
         # If this board uses the Amazon FreeRTOS reference bootlaoder, then we want to
         # build and flash the factory image.
-        if self._buildConfig.get('use_reference_bootloader', False):
+        if self._buildConfig.get('use_reference_bootloader', False) and OtaAfrProject.OTA_FACTORY_IMAGE_GENERATOR_PATH is not None:
```
Comment (azlwwan) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py]: "is not None" is not required.

Comment (HarnishaA) [reply tools/ota_e2e_tests/aws_ota_test/aws_ota_project.py]: Removed all of them in below commit. :-)

**Pull request merged**